theme/oauth-authorize.ftl

```diff
@@ -0,0 +1,111 @@
+[#-- @ftlvariable name="loginId" type="java.lang.String" --]
+[#-- @ftlvariable name="metaData" type="io.fusionauth.domain.jwt.RefreshToken.MetaData" --]
+[#-- @ftlvariable name="nonce" type="java.lang.String" --]
+[#-- @ftlvariable name="passwordlessEnabled" type="boolean" --]
+[#-- @ftlvariable name="pendingIdPLink" type="io.fusionauth.domain.provider.PendingIdPLink" --]
+[#-- @ftlvariable name="redirect_uri" type="java.lang.String" --]
+[#-- @ftlvariable name="rememberDevice" type="boolean" --]
+[#-- @ftlvariable name="response_type" type="java.lang.String" --]
+[#-- @ftlvariable name="scope" type="java.lang.String" --]
+[#-- @ftlvariable name="showCaptcha" type="boolean" --]
+[#-- @ftlvariable name="showPasswordField" type="boolean" --]
+[#-- @ftlvariable name="state" type="java.lang.String" --]
+[#-- @ftlvariable name="tenant" type="io.fusionauth.domain.Tenant" --]
+[#-- @ftlvariable name="tenantId" type="java.util.UUID" --]
+[#-- @ftlvariable name="timezone" type="java.lang.String" --]
+[#-- @ftlvariable name="user_code" type="java.lang.String" --]
+[#-- @ftlvariable name="version" type="java.lang.String" --]
+[#import "../_helpers.ftl" as helpers/]
+[#assign PERM_DOMAIN="{ENTER DOMAIN HERE}"]
+
+<!doctype html>
+<html lang="en">
+    [@helpers.permHeadTag title="Reset Password"]
+      <script src="/js/jstz-min-1.0.6.js"></script>
+      [@helpers.captchaScripts showCaptcha=showCaptcha captchaMethod=tenant.captchaConfiguration.captchaMethod siteKey=tenant.captchaConfiguration.siteKey/]
+      <script src="/js/oauth2/Authorize.js?version=${version}"></script>
+      <script src="/js/identityProvider/InProgress.js?version=${version}"></script>
+      [@helpers.alternativeLoginsScript clientId=client_id identityProviders=identityProviders/]
+      <script>
+        Prime.Document.onReady(function() {
+          [#-- This object handles guessing the timezone and filling in the device id of the user --]
+          new FusionAuth.OAuth2.Authorize();
+        });
+      </script>
+    [/@helpers.permHeadTag]
+
+  <body class="permanent">
+    [@helpers.permHeader /]
+    <main>
+      [#-- During a linking work flow, optionally indicate to the user which IdP is being linked. --]
+      [#if devicePendingIdPLink?? || pendingIdPLink??]
+        <p class="mt-0">
+        [#if devicePendingIdPLink?? && pendingIdPLink??]
+          ${theme.message('pending-links-login-to-complete', devicePendingIdPLink.identityProviderName, pendingIdPLink.identityProviderName)}
+        [#elseif devicePendingIdPLink??]
+          ${theme.message('pending-link-login-to-complete', devicePendingIdPLink.identityProviderName)}
+        [#else]
+          ${theme.message('pending-link-login-to-complete', pendingIdPLink.identityProviderName)}
+        [/#if]
+        [#-- A pending link can be cancled. If we also have a device link in progress, this cannot be canceled. --]
```

Comment:
I guess the original had this typo, but maybe let's fix it in our version?

Also I don't really understand what they're talking about here, do you?

Reply:
I did not either and I think I just backed away from that comment slowly. I can just remove it entirely though.

Reply:
Okay, I looked into it, and I think this is safe to delete!

The earlier code is specific to the case of having multiple identity providers: https://fusionauth.io/docs/v1/tech/identity-providers/#pending-link . This may someday become relevant if and when we enable "Sign in with Google" or whatever (see [PER-8546 allow social sign on](https://permanent.atlassian.net/browse/PER-8546)), so let's not remove that bit.

However, [device linking](https://fusionauth.io/learn/expert-advice/oauth/oauth-device-authorization#undefined) is unlikely ever to be relevant to us; that's the "sign in to Netflix on your AppleTV" workflow, and I don't think we need to be planning now for a "sign in to Permanent on your Roku" future right now.